javaslang/src/main/java/javaslang/collection/Traversable.java

```diff
@@ -228,8 +228,12 @@
      * @throws NullPointerException if {@code elements} is null
      */
     default boolean containsAll(Iterable<? extends T> elements) {
-        final HashSet<T> uniqueElements = HashSet.ofAll(elements);
-        return toSet().intersect(uniqueElements).size() == uniqueElements.size();
+        for (T element : elements) {
+            if (!contains(element)) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
A mutable j.u.List view ran in a StackOverflowError when a List was added to itself and checked if it containsAll() of its own elements.

This simplification fixes that problem.